File: dynamic_scripts/javascript/template.js
// *** This script will be reload automatically. So you can change your response dynamically.***
// *** dynamic-js-route is too slow. dynamic-python-route is fast.***
// the value at the end of this file will be return to client as a response.


const data = 'This is a sample data';
const end = ['!', '?', '.', '!!', '!?'];

// This value will be return as response.
// response type can be number, string, array, dictionary, boolean.
const response = {
    // type: 'number',
    // data: Math.random(),

    type: 'string',
    data: data + end[Math.floor(Math.random() * end.length)],

    // type: 'array',
    // data: end,

    // type: 'dictionary',
    // data: {a: 1, b: 2, c: 3, d: [1,2,3,4]},

    // type: 'boolean',
    // data: false,
}

File: dynamic_scripts/python/template.py
"""
*** This script will be reload automatically. So you can change your response dynamically.***
*** If this file raise a Exception (without SanicException) when reloading, system will use old data.***
The function named `moca` will be called, when received a request.
`moca` function must return a Sanic response.
Request object will be passed to `moca` function.
"""

# -- Imports --------------------------------------------------------------------------

from sanic.request import Request
from sanic.response import HTTPResponse, text
from random import choice

# -------------------------------------------------------------------------- Imports --

# -- Variables --------------------------------------------------------------------------

data = 'This is a sample data'
end = ['!', '?', '.', '!!', '!?']

# -------------------------------------------------------------------------- Variables --

# -- Moca --------------------------------------------------------------------------


def moca(request: Request) -> HTTPResponse:
    return text(data + choice(end))

# -------------------------------------------------------------------------- Moca --
